File: src/main/java/org/reflections/scanners/MethodAnnotationsScanner.java
package org.reflections.scanners;

import java.util.List;

/**
 * scans for method's annotations
 */
@SuppressWarnings("unchecked")
public class MethodAnnotationsScanner extends AbstractScanner {

    @Override
    public void scan(Object cls) {
        for (Object method : getMetadataAdapter().getMethods(cls)) {
            for (String methodAnnotation : (List<String>) getMetadataAdapter().getMethodAnnotationNames(method)) {
                if (acceptResult(methodAnnotation)) {
                    getStore().put(methodAnnotation, getMetadataAdapter().getMethodFullKey(cls, method));
                }
            }
        }
    }
}

File: src/main/java/org/reflections/scanners/MethodAnnotationsScanner.kt
package org.reflections.scanners

import org.reflections.ClassWrapper

/**
 * scans for method's annotations
 */
class MethodAnnotationsScanner : AbstractScanner() {

    override fun scan(cls: ClassWrapper) {
        for (method in metadataAdapter.getMethods(cls)) {
            for (methodAnnotation in metadataAdapter.getMethodAnnotationNames(method)) {
                if (acceptResult(methodAnnotation)) {
                    store!!.put(methodAnnotation, metadataAdapter.getMethodFullKey(cls, method))
                }
            }
        }
    }
}
